packages/react-vapor/src/components/actions/ActionBar.tsx

```diff
@@ -185,4 +185,11 @@ const mapDispatchToProps = (dispatch: IDispatch<IReactVaporState>, ownProps: IAc
     },
 });
 
-export const ActionBarConnected = connect(mapStateToProps, mapDispatchToProps)(ActionBar);
+export const ActionBarConnected = connect<
+    ReturnType<typeof mapStateToProps>,
+    ReturnType<typeof mapDispatchToProps>,
+    IActionBarProps
+>(
+    mapStateToProps,
+    mapDispatchToProps
+)(ActionBar as any);

```

Comment:
Why `as any` here?

Reply:
Both ActionBar and ActionBarConnected are used through the admin. If I don't cast `as any` I can't make the mapStateToProps and mapDispatchToProps props optional in the non-connected component. (They are optional in one interface and required in the other).

By casting to any, I make the compiler happy but since I added the generics to the variable, both ActionBar and ActionBarConnected have the correct props when they are used